[WebRTC] Refactor generic code of LibWebRTCProvider into new WebRTCProvider

philn opened the pull request:
#### cd335044c81d195337935e921f49bd6c59311c92
<pre>
[WebRTC] Refactor generic code of LibWebRTCProvider into new WebRTCProvider
<a href="https://bugs.webkit.org/show_bug.cgi?id=243387">https://bugs.webkit.org/show_bug.cgi?id=243387</a>

Reviewed by Carlos Garcia Campos.

This patch refactors the WebRTC backend agnostic code into a new WebRTCProvider base class. The
LibWebRTCProvider inherits from it. A GStreamerWebRTCProvider is also introduced, mostly empty for
now, future patches will enhance it.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/mediacapabilities/MediaCapabilities.cpp:
(WebCore::gatherDecodingInfo):
(WebCore::gatherEncodingInfo):
* Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp:
(WebCore::PeerConnectionBackend::PeerConnectionBackend):
(WebCore::PeerConnectionBackend::generateCertificate):
* Source/WebCore/Modules/mediastream/PeerConnectionBackend.h:
* Source/WebCore/Modules/mediastream/RTCController.cpp:
(WebCore::RTCController::disableICECandidateFilteringForAllOrigins):
(WebCore::RTCController::disableICECandidateFilteringForDocument):
(WebCore::RTCController::enableICECandidateFiltering):
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::RTCPeerConnection::create):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCPeerConnectionBackend.cpp:
(WebCore::createLibWebRTCPeerConnectionBackend):
(WebCore::PeerConnectionBackend::receiverCapabilities):
(WebCore::PeerConnectionBackend::senderCapabilities):
(WebCore::LibWebRTCPeerConnectionBackend::setConfiguration):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpReceiverBackend.cpp:
(WebCore::LibWebRTCRtpReceiverBackend::createSource):
* Source/WebCore/Modules/webaudio/AudioWorkletMessagingProxy.cpp:
(WebCore::AudioWorkletMessagingProxy::createRTCDataChannelRemoteHandlerConnection):
* Source/WebCore/Sources.txt:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::createRTCDataChannelRemoteHandlerConnection):
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::pageConfigurationWithEmptyClients):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::Page):
(WebCore::Page::settingsDidChange):
(WebCore::Page::applicationDidEnterBackground):
(WebCore::Page::applicationWillEnterForeground):
(WebCore::Page::configureLoggingChannel):
* Source/WebCore/page/Page.h:
(WebCore::Page::webRTCProvider):
(WebCore::Page::libWebRTCProvider): Deleted.
* Source/WebCore/page/PageConfiguration.cpp:
(WebCore::PageConfiguration::PageConfiguration):
* Source/WebCore/page/PageConfiguration.h:
* Source/WebCore/platform/GStreamer.cmake:
* Source/WebCore/platform/mediastream/WebRTCProvider.cpp: Added.
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::WebRTCProvider::setActive):
(WebCore::WebRTCProvider::setH265Support):
(WebCore::WebRTCProvider::setVP9Support):
(WebCore::WebRTCProvider::setVP9VTBSupport):
(WebCore::WebRTCProvider::codecCapability):
(WebCore::WebRTCProvider::disableEnumeratingAllNetworkInterfaces):
(WebCore::WebRTCProvider::enableEnumeratingAllNetworkInterfaces):
(WebCore::WebRTCProvider::audioDecodingCapabilities):
(WebCore::WebRTCProvider::videoDecodingCapabilities):
(WebCore::WebRTCProvider::audioEncodingCapabilities):
(WebCore::WebRTCProvider::videoEncodingCapabilities):
(WebCore::WebRTCProvider::computeVPParameters):
(WebCore::WebRTCProvider::createDecodingConfiguration):
(WebCore::WebRTCProvider::createEncodingConfiguration):
(WebCore::WebRTCProvider::videoDecodingCapabilitiesOverride):
(WebCore::WebRTCProvider::videoEncodingCapabilitiesOverride):
* Source/WebCore/platform/mediastream/WebRTCProvider.h: Copied from Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h.
(WebCore::WebRTCProvider::createRTCDataChannelRemoteHandlerConnection):
(WebCore::WebRTCProvider::isSupportingH265 const):
(WebCore::WebRTCProvider::isSupportingVP9Profile0 const):
(WebCore::WebRTCProvider::isSupportingVP9Profile2 const):
(WebCore::WebRTCProvider::isSupportingVP9VTB const):
(WebCore::WebRTCProvider::disableNonLocalhostConnections):
(WebCore::WebRTCProvider::isSupportingMDNS const):
(WebCore::WebRTCProvider::setLoggingLevel):
(WebCore::WebRTCProvider::setEnableWebRTCEncryption):
(WebCore::WebRTCProvider::setUseDTLS10):
(WebCore::WebRTCProvider::clearFactory):
(WebCore::WebRTCProvider::isEnumeratingAllNetworkInterfacesEnabled const):
(WebCore::WebRTCProvider::isVPSoftwareDecoderSmooth):
(WebCore::WebRTCProvider::isVPXEncoderSmooth):
(WebCore::WebRTCProvider::isH264EncoderSmooth):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.cpp: Added.
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed):
(WebCore::WebRTCProvider::create):
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::WebRTCProvider::setActive):
(WebCore::GStreamerWebRTCProvider::receiverCapabilities):
(WebCore::GStreamerWebRTCProvider::senderCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeAudioEncodingCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeVideoEncodingCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeAudioDecodingCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeVideoDecodingCapabilities):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.h: Added.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.cpp:
(WebCore::LibWebRTCProvider::LibWebRTCProvider):
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed):
(WebCore::WebRTCProvider::setActive):
(WebCore::computeLogLevel):
(WebCore::LibWebRTCProvider::createSocketFactory):
(WebCore::LibWebRTCProvider::startedNetworkThread):
(WebCore::LibWebRTCProvider::setPeerConnectionFactory):
(WebCore::LibWebRTCProvider::initializeAudioDecodingCapabilities):
(WebCore::LibWebRTCProvider::initializeVideoDecodingCapabilities):
(WebCore::LibWebRTCProvider::initializeAudioEncodingCapabilities):
(WebCore::LibWebRTCProvider::initializeVideoEncodingCapabilities):
(WebCore::LibWebRTCProvider::videoDecodingCapabilitiesOverride):
(WebCore::LibWebRTCProvider::videoEncodingCapabilitiesOverride):
(WebCore::LibWebRTCProvider::create): Deleted.
(WebCore::LibWebRTCProvider::webRTCAvailable): Deleted.
(WebCore::LibWebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
(WebCore::LibWebRTCProvider::setActive): Deleted.
(WebCore::LibWebRTCProvider::disableEnumeratingAllNetworkInterfaces): Deleted.
(WebCore::LibWebRTCProvider::enableEnumeratingAllNetworkInterfaces): Deleted.
(WebCore::LibWebRTCProvider::setH265Support): Deleted.
(WebCore::LibWebRTCProvider::setVP9Support): Deleted.
(WebCore::LibWebRTCProvider::setVP9VTBSupport): Deleted.
(WebCore::LibWebRTCProvider::audioDecodingCapabilities): Deleted.
(WebCore::LibWebRTCProvider::videoDecodingCapabilities): Deleted.
(WebCore::LibWebRTCProvider::audioEncodingCapabilities): Deleted.
(WebCore::LibWebRTCProvider::videoEncodingCapabilities): Deleted.
(WebCore::LibWebRTCProvider::codecCapability): Deleted.
(WebCore::computeVPParameters): Deleted.
(WebCore::isVPSoftwareDecoderSmooth): Deleted.
(WebCore::LibWebRTCProvider::createDecodingConfiguration): Deleted.
(WebCore::LibWebRTCProvider::createEncodingConfiguration): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
(WebCore::LibWebRTCProvider::audioModule):
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
(WebCore::WebRTCProvider::create):
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed):
(WebCore::WebRTCProviderCocoa::computeVPParameters):
(WebCore::WebRTCProviderCocoa::isVPSoftwareDecoderSmooth):
(WebCore::WebRTCProviderCocoa::setActive):
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::WebRTCProvider::registerWebKitVP9Decoder):
(WebCore::WebRTCProvider::registerWebKitVP8Decoder):
(WebCore::LibWebRTCProvider::create): Deleted.
(WebCore::LibWebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
(WebCore::LibWebRTCProviderCocoa::setActive): Deleted.
(WebCore::LibWebRTCProvider::webRTCAvailable): Deleted.
(WebCore::LibWebRTCProvider::registerWebKitVP9Decoder): Deleted.
(WebCore::LibWebRTCProvider::registerWebKitVP8Decoder): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.h:
* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/LibWebRTCProviderGStreamer.cpp:
(WebCore::WebRTCProvider::create):
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::LibWebRTCProvider::create): Deleted.
(WebCore::LibWebRTCProvider::webRTCAvailable): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::emulateRTCPeerConnectionPlatformEvent):
(WebCore::Internals::useMockRTCPeerConnectionFactory):
(WebCore::Internals::setEnumeratingAllNetworkInterfacesEnabled):
(WebCore::Internals::clearPeerConnectionFactory):
(WebCore::Internals::setWebRTCH265Support):
(WebCore::Internals::setWebRTCVP9Support):
(WebCore::Internals::setWebRTCVP9VTBSupport):
(WebCore::Internals::isSupportingVP9VTB const):
(WebCore::Internals::setEnableWebRTCEncryption):
(WebCore::Internals::setUseDTLS10):
(WebCore::Internals::setH264HardwareEncoderAllowed):
* Source/WebCore/workers/WorkerMessagingProxy.cpp:
(WebCore::WorkerMessagingProxy::createRTCDataChannelRemoteHandlerConnection):
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
(WebCore::ServiceWorkerThreadProxy::createRTCDataChannelRemoteHandlerConnection):
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.cpp:
(WebCore::SharedWorkerThreadProxy::createRTCDataChannelRemoteHandlerConnection):
* Source/WebKit/WebProcess/Network/NetworkProcessConnection.cpp:
(WebKit::NetworkProcessConnection::NetworkProcessConnection):
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
* Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.h:
* Source/WebKit/WebProcess/Storage/RemoteWorkerLibWebRTCProvider.h:
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::installServiceWorker):
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.cpp:
(WebKit::WebSharedWorkerContextManagerConnection::launchSharedWorker):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_appHighlightsVisible):

Canonical link: <a href="https://commits.webkit.org/253257@main">https://commits.webkit.org/253257@main</a>
</pre>

youennf commented on Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp:
```diff
@@ -446,7 +446,8 @@ void PeerConnectionBackend::generateCertificate(Document& document, const Certif
     }
 
-    LibWebRTCCertificateGenerator::generateCertificate(document.securityOrigin(), page->libWebRTCProvider(), info, [promise = WTFMove(promise)](auto&& result) mutable {
+    auto& webRTCProvider = reinterpret_cast<LibWebRTCProvider&>(page->webRTCProvider());
```
A static_cast should work to, given LibWebRTCProvider probably derives from WebRTCProvider.

youennf commented on Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCPeerConnectionBackend.cpp:
```diff
@@ -63,9 +63,10 @@ static std::unique_ptr<PeerConnectionBackend> createLibWebRTCPeerConnectionBacke
         return nullptr;
 
-    page->libWebRTCProvider().setEnableWebRTCEncryption(page->settings().webRTCEncryptionEnabled());
+    auto& webRTCProvider = reinterpret_cast<LibWebRTCProvider&>(page->webRTCProvider());
```
Ditto here and below

youennf commented on Source/WebCore/platform/mediastream/WebRTCProvider.h:
```diff
@@ -0,0 +1,137 @@
+
+    virtual void setLoggingLevel(WTFLogLevel) { }
+    virtual void setEnableWebRTCEncryption(bool) { }
+    virtual void setUseDTLS10(bool) { }
```
I wonder whether these might not be libwebrtc specific

philn replied:
I think the first might be doable in the GstWebRTC backend. I agree the two others are probably better off in the libwebrtc provider, I don't think those features are configurable in GstWebRTC.

youennf commented on Source/WebCore/platform/mediastream/WebRTCProvider.h:
```diff
@@ -0,0 +1,137 @@
+    bool isSupportingH265() const { return m_supportsH265; }
+    bool isSupportingVP9Profile0() const { return m_supportsVP9Profile0; }
+    bool isSupportingVP9Profile2() const { return m_supportsVP9Profile2; }
+    virtual bool isSupportingVP9VTB() const { return m_supportsVP9VTB; }
```
VTB is COCOA specific. Maybe move it back to LibWebRTC as well?

philn replied:
Right, I thought about it indeed :) the 3 others isSupporting methods are fine in the base class though, imho.

youennf commented on Source/WebCore/platform/mediastream/WebRTCProvider.h:
```diff
@@ -0,0 +1,137 @@
+    virtual void setActive(bool);
+
+    using IPAddressOrError = Expected<String, MDNSRegisterError>;
+    using MDNSNameOrError = Expected<String, MDNSRegisterError>;
```
These two do not seem used anymore, probably at the time we changed our handling of MDNS candidates

youennf commented on Source/WebCore/platform/mediastream/WebRTCProvider.h:
```diff
@@ -0,0 +1,137 @@
+    virtual bool isVPXEncoderSmooth(const VideoConfiguration&) { return false; }
+    virtual bool isH264EncoderSmooth(const VideoConfiguration&) { return true; }
+
+    bool m_enableEnumeratingAllNetworkInterfaces { false };
```
Might be libwebrtc specific.

youennf commented on Source/WebCore/platform/mediastream/WebRTCProvider.h:
```diff
@@ -0,0 +1,137 @@
+
+    bool m_enableEnumeratingAllNetworkInterfaces { false };
+    // FIXME: Remove m_useNetworkThreadWithSocketServer member variable and make it a global.
+    bool m_useNetworkThreadWithSocketServer { true };
```
This is definitely libwebrtc specific

youennf reviewed (commented):
Overall direction seems good to me.
I wonder whether we should not trim down a bit more the WebRTCProvider API.

philn committed:
[WebRTC] Refactor generic code of LibWebRTCProvider into new WebRTCProvider
https://bugs.webkit.org/show_bug.cgi?id=243387

Reviewed by Carlos Garcia Campos.

This patch refactors the WebRTC backend agnostic code into a new WebRTCProvider base class. The
LibWebRTCProvider inherits from it. A GStreamerWebRTCProvider is also introduced, mostly empty for
now, future patches will enhance it.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/mediacapabilities/MediaCapabilities.cpp:
(WebCore::gatherDecodingInfo):
(WebCore::gatherEncodingInfo):
* Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp:
(WebCore::PeerConnectionBackend::PeerConnectionBackend):
(WebCore::PeerConnectionBackend::generateCertificate):
* Source/WebCore/Modules/mediastream/PeerConnectionBackend.h:
* Source/WebCore/Modules/mediastream/RTCController.cpp:
(WebCore::RTCController::disableICECandidateFilteringForAllOrigins):
(WebCore::RTCController::disableICECandidateFilteringForDocument):
(WebCore::RTCController::enableICECandidateFiltering):
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::RTCPeerConnection::create):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCPeerConnectionBackend.cpp:
(WebCore::createLibWebRTCPeerConnectionBackend):
(WebCore::PeerConnectionBackend::receiverCapabilities):
(WebCore::PeerConnectionBackend::senderCapabilities):
(WebCore::LibWebRTCPeerConnectionBackend::setConfiguration):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpReceiverBackend.cpp:
(WebCore::LibWebRTCRtpReceiverBackend::createSource):
* Source/WebCore/Modules/webaudio/AudioWorkletMessagingProxy.cpp:
(WebCore::AudioWorkletMessagingProxy::createRTCDataChannelRemoteHandlerConnection):
* Source/WebCore/Sources.txt:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::createRTCDataChannelRemoteHandlerConnection):
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::pageConfigurationWithEmptyClients):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::Page):
(WebCore::Page::settingsDidChange):
(WebCore::Page::applicationDidEnterBackground):
(WebCore::Page::applicationWillEnterForeground):
(WebCore::Page::configureLoggingChannel):
* Source/WebCore/page/Page.h:
(WebCore::Page::webRTCProvider):
(WebCore::Page::libWebRTCProvider): Deleted.
* Source/WebCore/page/PageConfiguration.cpp:
(WebCore::PageConfiguration::PageConfiguration):
* Source/WebCore/page/PageConfiguration.h:
* Source/WebCore/platform/GStreamer.cmake:
* Source/WebCore/platform/mediastream/WebRTCProvider.cpp: Added.
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::WebRTCProvider::setActive):
(WebCore::WebRTCProvider::setH265Support):
(WebCore::WebRTCProvider::setVP9Support):
(WebCore::WebRTCProvider::setVP9VTBSupport):
(WebCore::WebRTCProvider::codecCapability):
(WebCore::WebRTCProvider::disableEnumeratingAllNetworkInterfaces):
(WebCore::WebRTCProvider::enableEnumeratingAllNetworkInterfaces):
(WebCore::WebRTCProvider::audioDecodingCapabilities):
(WebCore::WebRTCProvider::videoDecodingCapabilities):
(WebCore::WebRTCProvider::audioEncodingCapabilities):
(WebCore::WebRTCProvider::videoEncodingCapabilities):
(WebCore::WebRTCProvider::computeVPParameters):
(WebCore::WebRTCProvider::createDecodingConfiguration):
(WebCore::WebRTCProvider::createEncodingConfiguration):
(WebCore::WebRTCProvider::videoDecodingCapabilitiesOverride):
(WebCore::WebRTCProvider::videoEncodingCapabilitiesOverride):
* Source/WebCore/platform/mediastream/WebRTCProvider.h: Copied from Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h.
(WebCore::WebRTCProvider::createRTCDataChannelRemoteHandlerConnection):
(WebCore::WebRTCProvider::isSupportingH265 const):
(WebCore::WebRTCProvider::isSupportingVP9Profile0 const):
(WebCore::WebRTCProvider::isSupportingVP9Profile2 const):
(WebCore::WebRTCProvider::isSupportingVP9VTB const):
(WebCore::WebRTCProvider::disableNonLocalhostConnections):
(WebCore::WebRTCProvider::isSupportingMDNS const):
(WebCore::WebRTCProvider::setLoggingLevel):
(WebCore::WebRTCProvider::setEnableWebRTCEncryption):
(WebCore::WebRTCProvider::setUseDTLS10):
(WebCore::WebRTCProvider::clearFactory):
(WebCore::WebRTCProvider::isEnumeratingAllNetworkInterfacesEnabled const):
(WebCore::WebRTCProvider::isVPSoftwareDecoderSmooth):
(WebCore::WebRTCProvider::isVPXEncoderSmooth):
(WebCore::WebRTCProvider::isH264EncoderSmooth):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.cpp: Added.
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed):
(WebCore::WebRTCProvider::create):
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::WebRTCProvider::setActive):
(WebCore::GStreamerWebRTCProvider::receiverCapabilities):
(WebCore::GStreamerWebRTCProvider::senderCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeAudioEncodingCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeVideoEncodingCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeAudioDecodingCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeVideoDecodingCapabilities):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.h: Added.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.cpp:
(WebCore::LibWebRTCProvider::LibWebRTCProvider):
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed):
(WebCore::WebRTCProvider::setActive):
(WebCore::computeLogLevel):
(WebCore::LibWebRTCProvider::createSocketFactory):
(WebCore::LibWebRTCProvider::startedNetworkThread):
(WebCore::LibWebRTCProvider::setPeerConnectionFactory):
(WebCore::LibWebRTCProvider::initializeAudioDecodingCapabilities):
(WebCore::LibWebRTCProvider::initializeVideoDecodingCapabilities):
(WebCore::LibWebRTCProvider::initializeAudioEncodingCapabilities):
(WebCore::LibWebRTCProvider::initializeVideoEncodingCapabilities):
(WebCore::LibWebRTCProvider::videoDecodingCapabilitiesOverride):
(WebCore::LibWebRTCProvider::videoEncodingCapabilitiesOverride):
(WebCore::LibWebRTCProvider::create): Deleted.
(WebCore::LibWebRTCProvider::webRTCAvailable): Deleted.
(WebCore::LibWebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
(WebCore::LibWebRTCProvider::setActive): Deleted.
(WebCore::LibWebRTCProvider::disableEnumeratingAllNetworkInterfaces): Deleted.
(WebCore::LibWebRTCProvider::enableEnumeratingAllNetworkInterfaces): Deleted.
(WebCore::LibWebRTCProvider::setH265Support): Deleted.
(WebCore::LibWebRTCProvider::setVP9Support): Deleted.
(WebCore::LibWebRTCProvider::setVP9VTBSupport): Deleted.
(WebCore::LibWebRTCProvider::audioDecodingCapabilities): Deleted.
(WebCore::LibWebRTCProvider::videoDecodingCapabilities): Deleted.
(WebCore::LibWebRTCProvider::audioEncodingCapabilities): Deleted.
(WebCore::LibWebRTCProvider::videoEncodingCapabilities): Deleted.
(WebCore::LibWebRTCProvider::codecCapability): Deleted.
(WebCore::computeVPParameters): Deleted.
(WebCore::isVPSoftwareDecoderSmooth): Deleted.
(WebCore::LibWebRTCProvider::createDecodingConfiguration): Deleted.
(WebCore::LibWebRTCProvider::createEncodingConfiguration): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
(WebCore::LibWebRTCProvider::audioModule):
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
(WebCore::WebRTCProvider::create):
(WebCore::WebRTCProvider::setH264HardwareEncoderAllowed):
(WebCore::WebRTCProviderCocoa::computeVPParameters):
(WebCore::WebRTCProviderCocoa::isVPSoftwareDecoderSmooth):
(WebCore::WebRTCProviderCocoa::setActive):
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::WebRTCProvider::registerWebKitVP9Decoder):
(WebCore::WebRTCProvider::registerWebKitVP8Decoder):
(WebCore::LibWebRTCProvider::create): Deleted.
(WebCore::LibWebRTCProvider::setH264HardwareEncoderAllowed): Deleted.
(WebCore::LibWebRTCProviderCocoa::setActive): Deleted.
(WebCore::LibWebRTCProvider::webRTCAvailable): Deleted.
(WebCore::LibWebRTCProvider::registerWebKitVP9Decoder): Deleted.
(WebCore::LibWebRTCProvider::registerWebKitVP8Decoder): Deleted.
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.h:
* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/LibWebRTCProviderGStreamer.cpp:
(WebCore::WebRTCProvider::create):
(WebCore::WebRTCProvider::webRTCAvailable):
(WebCore::LibWebRTCProvider::create): Deleted.
(WebCore::LibWebRTCProvider::webRTCAvailable): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::resetToConsistentState):
(WebCore::Internals::emulateRTCPeerConnectionPlatformEvent):
(WebCore::Internals::useMockRTCPeerConnectionFactory):
(WebCore::Internals::setEnumeratingAllNetworkInterfacesEnabled):
(WebCore::Internals::clearPeerConnectionFactory):
(WebCore::Internals::setWebRTCH265Support):
(WebCore::Internals::setWebRTCVP9Support):
(WebCore::Internals::setWebRTCVP9VTBSupport):
(WebCore::Internals::isSupportingVP9VTB const):
(WebCore::Internals::setEnableWebRTCEncryption):
(WebCore::Internals::setUseDTLS10):
(WebCore::Internals::setH264HardwareEncoderAllowed):
* Source/WebCore/workers/WorkerMessagingProxy.cpp:
(WebCore::WorkerMessagingProxy::createRTCDataChannelRemoteHandlerConnection):
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
(WebCore::ServiceWorkerThreadProxy::createRTCDataChannelRemoteHandlerConnection):
* Source/WebCore/workers/shared/context/SharedWorkerThreadProxy.cpp:
(WebCore::SharedWorkerThreadProxy::createRTCDataChannelRemoteHandlerConnection):
* Source/WebKit/WebProcess/Network/NetworkProcessConnection.cpp:
(WebKit::NetworkProcessConnection::NetworkProcessConnection):
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
* Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.h:
* Source/WebKit/WebProcess/Storage/RemoteWorkerLibWebRTCProvider.h:
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::installServiceWorker):
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.cpp:
(WebKit::WebSharedWorkerContextManagerConnection::launchSharedWorker):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_appHighlightsVisible):

Canonical link: https://commits.webkit.org/253257@main
